### PR TITLE
mysql/decimal: fix a bug in do_div_mod when divisor/dividend is 0 

### PR DESCRIPTION
**Pull request by bb7133**
fix a bug in do_div_mod when divisor/dividend is 0 with fractional part

### CODE REVIEW

**Comment by bb7133** [comment]
@AndreMouche @breeswish PTAL, thanks!

File: src/coprocessor/codec/mysql/decimal.rs [hunk 1]
```diff
@@ -528,22 +528,45 @@ fn do_div_mod(
...
+            word_idx += 1;
+        }
+        if prec == 0 {
+            lhs.reset_to_zero();
+            return Some(Res::Ok(lhs));
+        } else {
+            prec -= count_leading_zeroes((prec - 1) % DIGITS_PER_WORD, lhs.word_buf[word_idx])
+        }
+        (word_idx, prec)
+    };
```
Comment (AndreMouche) [inline src/coprocessor/codec/mysql/decimal.rs]: Could you define L549~525 to an independent function?

**Comment by BusyJay** [comment]
@coocood PTAL

File: src/coprocessor/codec/mysql/decimal.rs (hunk 1 above)
Comment (bb7133) [reply src/coprocessor/codec/mysql/decimal.rs]: I don't get the semantic of the function: the code is a little bit tricky but very similar to remove_leading_zeros(), would we name the function something like remove_all_leading_zeros or calculate_idx_and_prec?

Comment (AndreMouche) [reply src/coprocessor/codec/mysql/decimal.rs]: Maybe you could transfer `lhs.int_cnt+l_frac_cnt` as a parameter for `remove_leading_zeroes` here?

**Comment by bb7133** [comment]
code updated, PTAL @AndreMouche , thanks

**Comment by coocood** [comment]
LGTM

@BusyJay PTAL

**Pull request merged**